### PR TITLE
Stack dns module, copy of 3.4.1 version module as such

### PR DESCRIPTION
**Pull request by bganapa**
## Description
Copy of the 3.4.1 module , changes only for the common run time dependency

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/StackAdmin/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj [hunk 1]
```diff
@@ -0,0 +1,371 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\..\..\..\tools\Common.Dependencies.Test.targets" />
```
Comment (maddieclayton) [inline src/StackAdmin/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj]: When do these tests get run?

File: src/StackAdmin/Dns/Commands.Dns.Test/packages.config [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
```
Comment (maddieclayton) [inline src/StackAdmin/Dns/Commands.Dns.Test/packages.config]: Take out the common packages here.

File: src/StackAdmin/Dns/Commands.Dns/AzureRM.Dns.psd1 [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+DotNetFrameworkVersion = '4.5.2'
+
+# Minimum version of the common language runtime (CLR) required by this module. This prerequisite is valid for the PowerShell Desktop edition only.
+CLRVersion = '4.0'
+
+# Processor architecture (None, X86, Amd64) required by this module
+# ProcessorArchitecture = ''
+
+# Modules that must be imported into the global environment prior to importing this module
+RequiredModules = @(@{ModuleName = 'AzureRM.Profile'; ModuleVersion = '5.3.3' })
```
Comment (maddieclayton) [inline src/StackAdmin/Dns/Commands.Dns/AzureRM.Dns.psd1]: Is this the version of Profile that you want to rely on? Just want to confirm.

File: src/StackAdmin/Dns/Commands.Dns/Commands.Dns.csproj [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    </EmbeddedResource>
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.Azure.Management.Dns, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Dns.1.7.2-preview\lib\net45\Microsoft.Azure.Management.Dns.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Configuration.Install" />
+  </ItemGroup>
+<!--  <ItemGroup>
```
Comment (maddieclayton) [inline src/StackAdmin/Dns/Commands.Dns/Commands.Dns.csproj]: Remove commented out lines.

File: src/StackAdmin/Dns/Dns.sln_ [hunk 5]
```diff
@@ -0,0 +1,106 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (maddieclayton) [inline src/StackAdmin/Dns/Dns.sln_]: Should the name of this file have an underscore at the end?

File: src/StackAdmin/Dns/Dns.sln_ [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Dns.Test", "Commands.Dns.Test\Commands.Dns.Test.csproj", "{133561EC-99AF-4ADC-AF41-39C4D3AD323B}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Common", "..\..\..\Common\Commands.Common\Commands.Common.csproj", "{5EE72C53-1720-4309-B54B-5FB79703195F}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Common.Authentication", "..\..\..\Common\Commands.Common.Authentication\Commands.Common.Authentication.csproj", "{D3804B64-C0D3-48F8-82EC-1F632F833C9E}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.ScenarioTests.ResourceManager.Common", "..\..\Common\Commands.ScenarioTests.ResourceManager.Common\Commands.ScenarioTests.ResourceManager.Common.csproj", "{3436A126-EDC9-4060-8952-9A1BE34CDD95}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Common.Authentication.Abstractions", "..\..\..\Common\Commands.Common.Authentication.Abstractions\Commands.Common.Authentication.Abstractions.csproj", "{70527617-7598-4AEF-B5BD-DB9186B8184B}"
```
Comment (maddieclayton) [inline src/StackAdmin/Dns/Dns.sln_]: Only include the necessary projects in here.

File: src/StackAdmin/Stack.sln [hunk 7]
```diff
@@ -65,16 +65,16 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Azs.Storage.Admin.Tests", "
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Profile", "..\ResourceManager\Profile\Commands.Profile\Commands.Profile.csproj", "{142D7B0B-388A-4CEB-A228-7F6D423C5C2E}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Compute", "Compute\Commands.Compute\Commands.Compute.csproj", "{52643BD5-6378-49BD-9F6E-DAC9DD8A867B}"
```
Comment (maddieclayton) [inline src/StackAdmin/Stack.sln]: Why are you removing Compute and Network?

**Comment by maddieclayton** [comment]
@bganapa Also looks like you have a bunch of test failures that for some reason are not failing the build.  See 01:57:30 here: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/6914/consoleFull

**Comment by maddieclayton** [comment]
@deathly809 Got it. Are you planning to make these tests fail the build at some point? It took digging into the log to even discover that they were failing, so in the future anyone could break stack and not even realize.

**Comment by deathly809** [comment]
@maddieclayton Mark mentioned to use that we should not depend on the modules themselves for tests but use some common code that everyone can use.  It would take a lot of work though and would be a future change.

File: src/StackAdmin/Dns/Commands.Dns.Test/packages.config (hunk 2 above)
Comment (bganapa) [reply src/StackAdmin/Dns/Commands.Dns.Test/packages.config]: done

File: src/StackAdmin/Dns/Commands.Dns/Commands.Dns.csproj (hunk 4 above)
Comment (bganapa) [reply src/StackAdmin/Dns/Commands.Dns/Commands.Dns.csproj]: done

File: src/StackAdmin/Dns/Dns.sln_ (hunk 6 above)
Comment (bganapa) [reply src/StackAdmin/Dns/Dns.sln_]: removed the .sln

File: src/StackAdmin/Stack.sln (hunk 7 above)
Comment (bganapa) [reply src/StackAdmin/Stack.sln]: Compute and network will be added as a seperate PR

File: src/StackAdmin/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE;SIGN</DefineConstants>
+    <Optimize>true</Optimize>
+    <DebugType>pdbonly</DebugType>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <ErrorReport>prompt</ErrorReport>
+    <RunCodeAnalysis>false</RunCodeAnalysis>
+  </PropertyGroup>
+  <ItemGroup>
+<!--    <Reference Include="Hyak.Common">
```
[the remark this answers is not in the document]
Comment (bganapa) [reply src/StackAdmin/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj]: done

File: src/StackAdmin/Dns/Commands.Dns/AzureRM.Dns.psd1 (hunk 3 above)
Comment (bganapa) [reply src/StackAdmin/Dns/Commands.Dns/AzureRM.Dns.psd1]: changed to 5.5.1

File: src/StackAdmin/Dns/Dns.sln_ (hunk 5 above)
Comment (bganapa) [reply src/StackAdmin/Dns/Dns.sln_]: removed this file

**Comment by bganapa** [comment]
ps-demand - https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/1081/
ps-sign - https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/1158/
psnetsign - https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/152/

**Comment by maddieclayton** [comment]
@bganapa This looks good to me, I will approve.  @deathly809 Please also approve.  Can this be merged before Storage or does it need to wait for the Storage PR to get in?

**Comment by bganapa** [comment]
@maddieclayton This can be merged before storage. Either one will result in a conflict in stack.sln :)

**Comment by maddieclayton** [comment]
Merged! Please update the Storage PR when you get a chance.

**Pull request merged**